app/models/challenge.rb

```diff
@@ -0,0 +1,42 @@
+class Challenge < ActiveRecord::Base
+  belongs_to :challenger, class_name: "Photo"
+  belongs_to :challenged, class_name: "Photo"
+
+  before_create :default_length
+  before_create :complete_false
+  after_update :results_email
+
+  def self.send_results
+    Challenge.where("ends_at < ? AND completed = ?", Time.now, false).each do |challenge|
+      challenge.complete!
+    end
+  end
+
+  private
+
+  def default_length
+    self.length ||= 6
+  end
+
+  def complete_false
+    self.complete = false
+    true
+  end
+
+  def results_email
+    if self.complete?
+      photo_challenger = self.challenger
+      photo_challenged = self.challenged
+      if self.votes_challenger == self.votes_challenged
+        ChallengeMailer.results_email(photo_challenger, 'tie')
```

Comment:
What if we had a method called "winner", it returns the challenger if they win, or the challenged if they win, or nil if it was a draw.

You could have the opposite of this method as well, called "loser".

You can also have a method called "draw?". That will return true if it was a tie.

This block becomes much cleaner:

```ruby
if draw?
  ChallengeMailer.results_email(photo_challenged, 'tie')
else
  ChallengeMailer.results_email(winner, 'win')
  ChallengeMailer.results_email(looser, 'lose')
end
```